athena-neptune/src/main/java/com/amazonaws/athena/connectors/neptune/NeptuneRecordHandler.java

```diff
@@ -111,55 +95,22 @@ protected NeptuneRecordHandler(final AmazonS3 amazonS3, final AWSSecretsManager
      *       performance.
      */
     @Override
-    protected void readWithConstraint(final BlockSpiller spiller, final ReadRecordsRequest recordsRequest,
-            final QueryStatusChecker queryStatusChecker) throws Exception 
+    protected void readWithConstraint(final BlockSpiller spiller, final ReadRecordsRequest recordsRequest, final QueryStatusChecker queryStatusChecker) throws Exception 
     {
         logger.info("readWithConstraint: enter - " + recordsRequest.getSplit());
-        TableName tableName = recordsRequest.getTableName();
-        String labelName = tableName.getTableName();
-        long numRows = 0;
         Client client = null;
-        GraphTraversalSource graphTraversalSource = null;
+        String graphType = System.getenv("neptune_graphtype") != null ? System.getenv("neptune_graphtype") : "PROPERTYGRAPH";
```

Comment:
Is there a way to move this decision into the MetadataHandler's split creation logic and then this code can just look it up from the split? The reason I ask is because in the future relying on the env to be present for random pieces of code will be tricky. It is in our best interest to limit where and how we use getEnv(...)

Reply:
This parameter is from setup and is a mandatory field. One deployment will either have Property graph or RDF graph